### PR TITLE
 Increased accurate in regards to recording proper delays between act…

### PR DESCRIPTION
**Pull request by ra314**
…ions.

https://github.com/RMPR/atbswp/issues/37#issuecomment-1061614623

### CODE REVIEW

**Review by RMPR — COMMENTED** [review]
Awesome work, can you have a look at the suggested changes?

**Comment by ra314** [comment]
Thanks.
I applied the changes.
Just curious, were the changes purely for readability purposes.
i.e to make it clear that timeout is a flaot?
Cause these lines are otherwise equivalent right?

**Comment by RMPR** [comment]
These lines might be equivalent to the user, but for the Python interpreter they are not because [comparison is a pain](https://github.com/python/cpython/blob/850687df47b03e98c1433e6e70e71a8921eb4454/Objects/floatobject.c#L376).

`timeout >= 0` is equivalent to `timeout.__gt__(0)`

Now, you need to consider two things:

- When you explicitly convert 0 to 0.0, the comparison is done as straightforwardly (and long-windedly) as conceivable. It's done so that, e.g., Python x == y delivers the same result as the platform C x == y.

- When you don't explicitly convert into to a float before the comparison: `__gt__` needs to handle the conversion to float first (not as straightforward as one may think at first) then make the comparison.

 The maintainers say it much better than I ever could:

 > When mixing float with an integer type, there's no good *uniform* approach. Converting the double to an integer obviously doesn't work, since we may lose info from fractional bits.  Converting the integer to a double also has two failure modes:  (1) an int may trigger overflow (too large to fit in the dynamic range of a C double); (2) even a C long may have more bits than fit in a C double (e.g., on a 64-bit box long may have 63 bits of precision, but a C double probably has only 53), and then we can falsely claim equality when low-order integer bits are lost by coercion to double.  So this part is painful too.

**Review by RMPR — COMMENTED** [review]
Can you please [squash your commits into one](https://medium.com/@reyhansofian/squash-multiple-git-commit-into-one-commit-1d998dedc0da) and I'll happily merge.

**Comment by ra314** [comment]
Done. That was quite an interesting read, thanks for the information!

File: atbswp/control.py [hunk 1]
```diff
@@ -265,8 +266,8 @@ def on_scroll(self, x, y, dx, dy):
     def on_press(self, key):
         """Triggered by a key press."""
         b = time.perf_counter()
-        timeout = int(b - self.last_time)
-        if timeout > 0:
+        timeout = float(b - self.last_time)
+         if timeout > 0.0:
```
Comment (RMPR) [inline atbswp/control.py:270]: Unecessary extra space here.

File: atbswp/control.py [hunk 2]
```diff
@@ -221,8 +222,8 @@ def on_move(self, x, y):
         if not self.recording:
             return False
         b = time.perf_counter()
-        timeout = int(b - self.last_time)
-        if timeout > 0:
+        timeout = float(b - self.last_time)
+         if timeout > 0.0:
```
Comment (RMPR) [inline atbswp/control.py:226]: Unecessary extra space here.

File: atbswp/control.py [hunk 3]
```diff
@@ -39,6 +39,7 @@
 import wx.adv
 import wx.lib.newevent as NE
 
+pyautogui.PAUSE = 0
```
Comment (RMPR) [inline atbswp/control.py:42]: This doesn't belong here.

Comment (ra314) [reply atbswp/control.py]: I think I've moved it to the right place, but tbh it's just a hunch.

**Comment by ra314** [comment]
Sorry about that, idk how that space slipped by.

**Pull request merged**